components/ModSubmissionListContainer.vue

```diff
@@ -141,17 +143,27 @@ const submissionListItemTableColumns = computed(() => {
 })
 
 const handleClickToSubmissionForm = (id: string) => {
-    useModerationScreenStore().setActiveScreen(ModerationScreen.EditSubmission)
     modSubmissionsListStore.selectedSubmissionId = id
+    moderationScreenStore.setActiveScreen(ModerationScreen.EditSubmission)
 }
 
 const handleClickToHealthcareProfessionalForm = (id: string) => {
-    useModerationScreenStore().setActiveScreen(ModerationScreen.EditHealthcareProfessional)
-    modSubmissionsListStore.selectedHealthcareProfessionalId = id
+    healthcareProfessionalsStore.selectedHealthcareProfessionalId = id
+    moderationScreenStore.setActiveScreen(ModerationScreen.EditHealthcareProfessional)
 }
 
 const handleClickToFacilityForm = (id: string) => {
-    useModerationScreenStore().setActiveScreen(ModerationScreen.EditFacility)
-    modSubmissionsListStore.selectedFacilityId = id
+    facilitiesStore.selectedFacilityId = id
+    facilitiesStore.setSelectedFacilityData(id)
+    moderationScreenStore.setActiveScreen(ModerationScreen.EditFacility)
 }
+
+/** This ensures that the data for the selected facility is loaded before mounting the component,
+    thus avoiding a flash in the ui loading the refs**/
+onBeforeRouteLeave((to, _, next) => {
```

Comment:
while this works, it would be better to have an implementation on the page itself for two reasons: 
1) Some edge cases will circumvent this. Ex: what if a moderator wants  to send a link to another moderator directly to that facility? This wouldn't run in this case
2) When you read the facility page, you don't realize that there's logic outside of the page here that affects it. Ideally you  don't want related logic outside of a component.

Reply:
on the page, I believe we could use beforeMount or something similar. Basically set it before the render is displayed